cypress/e2e/audio-player/audio-player.spec.ts

```diff
@@ -0,0 +1,588 @@
+        cy.get(".mx_MainSplit .mx_ScrollPanel").scrollTo("bottom");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            checkPlayerFilenameLong();
+
+            // Enable IRC layout
+            cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.IRC);
+
+            // Click the timestamp to highlight the event tile in case it is not visible
+            cy.get(".mx_EventTile_last[data-layout='irc'] .mx_MessageTimestamp").click();
+
+            // Assert that rendering of the player settled and the play button is visible before taking a snapshot
+            cy.get(".mx_MAudioBody [data-testid='play-pause-button'][aria-label='Play']").should("be.visible");
+
+            cy.get(".mx_MAudioBody").percySnapshotElement("Audio player (light theme) on IRC layout", {
+                percyCSS,
+                widths: snapshotWidthsIRC,
+            });
+
+            // Output a log for reference/debugging
+            cy.log("Took a snapshot of Audio player (light theme) on IRC layout");
+        });
+    });
+
+    it("should be correctly rendered on modern and bubble layouts", () => {
+        visitRoom();
+
+        // Upload one second audio file with a long file name
+        uploadFile("cypress/fixtures/1sec-long-name-audio-file.ogg");
+
+        // Scroll to the bottom to make the audio player visible for Percy tests
+        cy.get(".mx_MainSplit .mx_ScrollPanel").scrollTo("bottom");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            checkPlayerFilenameLong();
+
+            // Reset to the default layout
+            cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Group);
+
+            // Take snapshots (light theme)
+            takeSnapshots("Audio player (light theme)");
+
+            // Take snapshots (light theme, monospace font)
+            cy.setSettingValue("useSystemFont", null, SettingLevel.DEVICE, true);
+            cy.setSettingValue("systemFont", null, SettingLevel.DEVICE, "monospace");
+            // Assert that the monospace timer is visible
+            cy.get("[role='timer']").should("have.css", "font-family", '"monospace"').should("be.visible");
+            takeSnapshots("Audio player (light theme, monospace)");
+
+            // Reset font setting
+            cy.setSettingValue("useSystemFont", null, SettingLevel.DEVICE, false);
+
+            // Take snapshots (dark theme)
+            cy.setSettingValue("theme", null, SettingLevel.ACCOUNT, "dark");
+            takeSnapshots("Audio player (dark theme)");
+        });
+    });
+
+    it("should be correctly rendered on high contrast theme", () => {
+        visitRoom();
+
+        // Upload one second audio file with a long file name
+        uploadFile("cypress/fixtures/1sec-long-name-audio-file.ogg");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            checkPlayerFilenameLong();
+        });
+
+        // Disable "Match system theme" in case
```

Comment:
... in case of what?

Reply:
I have spotted locally on Cypress once that the test could not enable the light theme by clicking as the option was grayed out probably because "use_system_theme" had been enabled. It was intermittent and is not consistently reproduced, so I am going to remove this. Let's re-add this if the test becomes flaky :crossed_fingers:

Reply:
This was removed with https://github.com/matrix-org/matrix-react-sdk/pull/10441/commits/1b279fd58f53cca0f41d6486c70bb2044591d4e6.

Reply:
> I have spotted locally on Cypress once that the test could not enable the light theme by clicking as the option was grayed out probably because "use_system_theme" had been enabled. It was intermittent and is not consistently reproduced

It was reproduced as below: https://cloud.cypress.io/projects/ppvnzg/runs/11488/test-results/f7fc4f91-d5f2-4b34-979f-b180d4689793. Let me re-add this line!

Reply:
no problem, but please could you give it a clearer comment?

Reply:
I see. I am going to add a better comment there :-)